app/products.py

```diff
@@ -0,0 +1,114 @@
+"""
+ Must have the environment variables set properly prior to running the script
+"""
+from woocommerce import API
+import json
+import os
+from faker import Faker
+import random
+
+fake = Faker()
+
+
+def create_api_object():
+    wcapi = API(
+        url=os.environ.get('BASE_URL'),
+        consumer_key=os.environ.get('API_KEY'),
+        consumer_secret=os.environ.get('API_SECRET'),
+        version="wc/v3",
+        timeout=15
+    )
+    return wcapi
+
+
+store_api = create_api_object()
+
+
+class Products:
+    def __init__(self) -> None:
+        self.endpoint = 'products'
+        self.products = self.fetch_all_products()
+
+    def create_random_product(self):
+        product_data = self.generate_random_product_data()
+
+        try:
+            response = store_api.post(self.endpoint, data=product_data)
+            if response.ok:
+                print("A product has been added successfully")
+            else:
+                raise Exception(f"Failed to add product. Status code: {response.status_code}")
+        except Exception:
+            print("Something went wrong while adding a product.")
```

Comment:
Print the exception itself too. that helps identify what the problem is. Otherwise user will have to run the script and remove the except block. No way of knowing what happened. So do this instead.
```
except Exception as e:
            print(f"Something went wrong while adding a product. Exception: {e}")
```